### PR TITLE
main.go: add more TLS options

### PR DESCRIPTION
**Pull request by dkropachev**
Adds:
      -tls-server-name option to enable server hostname verfication
      -tls-host-verification option to enable server hostname and certificate verification
      -tls-ca-cert-file option to enable server certificate verification
      -tls-client-cert-file option to enable client certificate specification
      -tls-client-key-file option to enable client key specification

    Refs: #37

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -207,6 +208,11 @@ func main() {
 		replicationFactor int
 
 		nodes             string
+		caCertFile		  string
```
Comment (piodul) [inline main.go:211]: Tabs seem to be mixed with spaces in this line. Please use spaces between `caCertFile` and `string`.

File: main.go [hunk 2]
```diff
@@ -373,6 +379,50 @@ func main() {
 		}
 	}
 
+	if tlsEncryption {
+		sslOpts := &gocql.SslOptions{
+			Config:&tls.Config{
```
Comment (piodul) [inline main.go:384]: Please add a single space after `Config:`

File: main.go [hunk 3]
```diff
@@ -373,6 +379,50 @@ func main() {
...
+		if clientCertFile != "" {
+			if _, err := os.Stat(clientCertFile); err != nil {
+				log.Fatal(err)
+			}
+			sslOpts.CertPath = clientCertFile
+		}
+
+		if clientKeyFile != "" || clientCertFile != "" {
+			if clientKeyFile == "" || clientCertFile == "" {
+				log.Fatal("ssl-client-key-file and ssl-client-cert-file should be provided side by side")
```
Comment (piodul) [inline main.go:413]: I think that this error message could be formulated better. "side by side" suggests to me that both options should be near each other in some way (next to each other as commandline arguments? or maybe key and cert file need to be in the same catalogue?).

How about doing two checks and splitting the error message like this:
```go
if clientKeyFile != "" && clientCertFile == "" {
    log.Fatal("ssl-client-cert-file is required when ssl-client-key-file is provided")
}
if clientCertFile != "" && clientKeyFile == "" {
    log.Fatal("ssl-client-key-file is required when ssl-client-cert-file is provided")
}
```

File: main.go [hunk 4]
```diff
@@ -262,8 +267,12 @@ func main() {
...
 
-	flag.BoolVar(&tls, "tls", false, "use TLS encryption")
-
+	flag.BoolVar(&tlsEncryption, "tls", false, "use TLS encryption")
+	flag.StringVar(&serverName, "tls-server-name", "", "TLS server hostname")
+	flag.BoolVar(&hostVerification, "tls-host-verification", false, "verify server certificate")
+	flag.StringVar(&caCertFile, "tls-ca-cert-file", "", "path to CA certificate file, needed to enable encryption")
+	flag.StringVar(&clientCertFile, "tls-client-cert-file", "", "path to client certificate file, needed to enable client certificate authentication")
+	flag.StringVar(&clientKeyFile, "tls-client-key-file", "", "path to client key file, needed to enable client certificate authentication")
 	flag.StringVar(&hostSelectionPolicy, "host-selection-policy", "token-aware", "set the driver host selection policy (round-robin,token-aware,dc-aware),default 'token-aware'")
```
Comment (piodul) [inline main.go:276]: Nit: add a blank line before the line with the `hostSelectionPolicy` parameter. The `hostSelectionPolicy` parameter is unrelated from TLS options, so it makes sense to separate it visually.

**Review by piodul — CHANGES_REQUESTED** [review]
Left some nitpicks, LGTM after fixing them.

File: main.go (hunk 2 above)
Comment (dkropachev) [reply main.go:384]: fixed

File: main.go (hunk 1 above)
Comment (dkropachev) [reply main.go:211]: fixed

File: main.go (hunk 4 above)
Comment (dkropachev) [reply main.go:277]: done

**Review by piodul — APPROVED** [review]

**Comment by dkropachev** [comment]
@mmatczuk , could you please merge it ?

**Review by mmatczuk — APPROVED** [review]

**Pull request merged**